app/sql.js

```diff
@@ -2220,6 +2280,34 @@ function getFirstUnreadMessageIdInConversation(conversationId) {
   return rows[0].id;
 }
 
+/**
+ * Deletes all but the 10,000 last received messages.
+ */
+function trimMessages() {
+  globalInstance
+    .prepare(
+      `
+      DELETE FROM ${MESSAGES_TABLE}
+      WHERE id NOT IN (
+        SELECT id FROM ${MESSAGES_TABLE}
+        ORDER BY received_at DESC
+        LIMIT 10000
+      );
+    `
+    )
+    .run();
+
+  const rows = globalInstance
+    .prepare(
+      `SELECT * FROM ${MESSAGES_TABLE}
+     ORDER BY received_at DESC;`
+    )
+    .all();
+
+  return rows;
+  // return map(rows, row => jsonToObject(row.json));
```

Comment:
this has to be on a per conversation basis. Like keep the last 1k messages on each conversation

Reply:
be sure to update the localized string too